physicsnemo/mesh/mesh.py

```diff
@@ -0,0 +1,1044 @@
+        number of points and :math:`D_s` is the spatial dimension. Must be a
+        floating-point dtype.
+    cells : torch.Tensor
+        Cell connectivity with shape :math:`(N_c, D_m + 1)` where :math:`N_c` is
+        the number of cells and :math:`D_m` is the manifold dimension. Each row
+        contains indices into ``points`` defining one simplex. Must be an integer
+        dtype, since these are treated as indices into the ``points`` tensor.
+    point_data : TensorDict or dict[str, torch.Tensor] (optional)
+        Per-vertex data. If a dict, converted to TensorDict with batch size
+        :math:`(N_p,)`. Default is an empty TensorDict.
+    cell_data : TensorDict or dict[str, torch.Tensor] (optional)
+        Per-cell data. If a dict, converted to TensorDict with batch size
+        :math:`(N_c,)`. Default is an empty TensorDict.
+    global_data : TensorDict or dict[str, torch.Tensor] (optional)
+        Mesh-level data. If a dict, converted to TensorDict with scalar batch
+        size. Default is an empty TensorDict.
+
+    Attributes
+    ----------
+    points : torch.Tensor
+        Vertex coordinates, shape :math:`(N_p, D_s)`.
+    cells : torch.Tensor
+        Cell connectivity, shape :math:`(N_c, D_m + 1)`.
+    point_data : TensorDict
+        Per-vertex data container with batch size :math:`(N_p,)`.
+    cell_data : TensorDict
+        Per-cell data container with batch size :math:`(N_c,)`.
+    global_data : TensorDict
+        Mesh-level data container with scalar batch size.
+
+    Raises
+    ------
+    ValueError
+        If ``points`` is not 2D, ``cells`` is not 2D, or manifold dimension
+        exceeds spatial dimension.
+    TypeError
+        If ``cells`` has a floating-point dtype (must be integer).
+
+    Examples
+    --------
+    Create a single triangle in 2D:
+
+    >>> import torch
+    >>> from physicsnemo.mesh import Mesh
+    >>> points = torch.tensor([[0.0, 0.0], [1.0, 0.0], [0.5, 1.0]])
+    >>> cells = torch.tensor([[0, 1, 2]])
+    >>> mesh = Mesh(points=points, cells=cells)
+    >>> mesh.n_points, mesh.n_cells, mesh.n_spatial_dims, mesh.n_manifold_dims
+    (3, 1, 2, 2)
+
+    Create a triangle mesh with attached scalar data:
+
+    >>> mesh = Mesh(
+    ...     points=points,
+    ...     cells=cells,
+    ...     point_data={"temperature": torch.tensor([300.0, 350.0, 325.0])},
+    ...     cell_data={"pressure": torch.tensor([101.3])},
+    ... )
+
+    Move mesh and all data to GPU:
+
+    >>> mesh_gpu = mesh.to("cuda")  # doctest: +SKIP
+
+    Notes
+    -----
+    **Simplex Definition**
+
+    An n-simplex is the convex hull of n+1 affinely independent points. The vertices
+    are stored as indices into the ``points`` array, so a triangle (2-simplex) is
+    represented by 3 indices, a tetrahedron (3-simplex) by 4 indices, etc.
+
+    **Caching**
+
+    Expensive geometric computations (centroids, areas, normals) are automatically
```

Comment:
While caching is nice, I'm wondering if there needs to be some controls since these could be fairly large tensors for big meshes.

Like I'm wondering if these should have setters (maybe this data could be manually loaded), getters (property) and deleter to handle the cache interface under the hood.